### PR TITLE
Sprint c02   undo claim announcement

### PR DESCRIPTION
**Pull request by DDanielVieira**

### CODE REVIEW

File: src/main/java/com/criticalsoftware/Announcement.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package com.criticalsoftware;
+
+import org.bson.types.ObjectId;
+
+import java.time.LocalDateTime;
+
+public class Announcement {
+
+    private String id;
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/Announcement.java:9]: As the id field is already inherited from the parent class (PanacheMongoEntity), it doesn't need to be explicitly declared in this class.

**Review by jrromao — COMMENTED** [review]
You can use lombok to avoid the explicit declaration of the constructors, getters and setters.

File: src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+    public AnnouncementResponse(String id, Product product, String userDonorId,String userDoneeId) {
+
+    }
+
+    public AnnouncementResponse(String id, Product product, LocalDateTime date, ObjectId userDonorId, ObjectId userDoneeId) {
+        this.id = id;
+        this.product = product;
+        this.date = LocalDateTime.now();
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java:25]: You should set the date's value to what is received by input.

File: src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    private String id;
+    private Product product;
+    private LocalDateTime date;
+    private ObjectId userDonorId;
+    private ObjectId userDoneeId;
+
+
+    public AnnouncementResponse(String id, Product product, String userDonorId,String userDoneeId) {
+
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java:19]: Why do you need this empty constructor?

File: src/main/java/com/criticalsoftware/Product.java [hunk 4]
```diff
@@ -1,23 +1,33 @@
...
-    private BufferedImage photo;
+    private String photo;
     private String category;
-    private User user;
 
     public Product() {
     }
 
-    public Product(String description, BufferedImage photo, String category, User user) {
+    public Product(String name,String description, String photo, String category, User userDonnor) {
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/Product.java:17]: Why do you have the input parameter "userDonnor" if you don't use it in the constructor?

**Review by jrromao — CHANGES_REQUESTED** [review]
You can use lombok to avoid the explicit declaration of the constructors, getters and setters.

File: src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java (hunk 3 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java:19]: If I remove the constructor the code will show errors

File: src/main/java/com/criticalsoftware/Product.java (hunk 4 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/Product.java:17]: done.

**Pull request merged**